mailbox/store/src/test/java/org/apache/james/mailbox/store/search/AbstractMessageSearchIndexTest.java

```diff
@@ -1143,4 +1146,30 @@ public void searchShouldReturnEmptyWhenMailboxBelongingToAnotherMailboxSession()
 
         assertThat(messageSearchIndex.search(otherSession, mailbox, searchQuery)).isEmpty();
     }
+
+    @Test
+    public void searchShouldOrderByInternalDateWhenSortOnSentDateAndNoCorrespondingHeader() throws Exception {
```

Comment:
Moreover I think one test is not enough we need three test:
 * none of the message has the header
 * one of them have it
 * all of them have it